### DISCUSSION TITLE
How to do Migration when using pocket base as framework?

### DISCUSSION THREAD

**Discussion by iamdineshkumar** [pocketbase/pocketbase#2218] [Q&A]
I am using pocket base as framework and during development migration folder is created and its getting update for changes i made. But when i deploy to server i am having migration folder next to build executable but its not migrating. Guide me on this.

**Comment by ganigeorgiev** [comment]
Please check https://pocketbase.io/docs/migrations/#enable-go-migrations.

You don't have to deploy the `migrations` directory when using Go migrations, because they will be embedded in the final executable.

**Reply by iamdineshkumar** [reply]
I used https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go as base for writing app. It actually generated pb_migration folder but when creating build and executing it. migration is not happening. Is there anything need to be imported like folder or whether anything need to be added?

**Reply by ganigeorgiev** [reply]
The example main.go file uses JS migrations because the migrations are external, aka. not part of the executable.

If you are using PocketBase as framework you can use directly the Go API. Once again, please make sure to check https://pocketbase.io/docs/migrations/#enable-go-migrations.

**Reply by iamdineshkumar** [reply]
oh thats the issue I belive... I will change to js. Since migration is not working, as a work around i just copy json from export in local and imported it in production server. So if i correct this now to Go and released changes to production server whether any breaking will happen?

**Reply by iamdineshkumar** [reply]
![image](https://user-images.githubusercontent.com/14805534/230112866-25dce888-842a-4da2-a26f-8f7c876457ab.png)
I got this error as well for ur attention.

**Reply by ganigeorgiev** [reply]
@iamdineshkumar `history-sync` was added in [PocketBase v0.14.0](https://github.com/pocketbase/pocketbase/releases/tag/v0.14.0). I don't understand what do you mean by "migration is not working". Could you provide a minimal reproducible repo?

Please note that the `example/base/main.go` is used for the prebuilt executable. You don't have to copy it 1:1. The minimal PocketBase `main.go` could be found in https://pocketbase.io/docs/use-as-framework/.

If you still for some reason want to load JS migration for your custom binary, you have to enable also the `jsvm` migrations plugin as shown in https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go#L70-L73.

But again, for most cases this shouldn't be necessary. Please make sure to read https://pocketbase.io/docs/migrations/#enable-go-migrations. If something is not clear from the linked guide, please elaborate what are the confusing parts so that I can update it and try to explain it better.

**Reply by iamdineshkumar** [reply]
Hi thanks for the information.
To make you understand my situation. I have already manually imported collections using provided export & import option under settings. So production is there with necessary table structure. If I am going to enable migration whether it will have impact in production server is my question.(Since already table are manually imported and migration will again try to create collection which is already exist)
